tests/test_cli_compile.py

```diff
@@ -2457,6 +2447,40 @@ def test_triple_equal_pinned_dependency_is_used(
 )
 
 
+@pytest.mark.network
+@pytest.mark.parametrize(("fname", "content"), METADATA_TEST_CASES)
+def test_not_specified_input_file(
+    fake_dists, runner, make_module, fname, content, monkeypatch
+):
+    """
+    Test that a default-named file is parsed if present.
+    """
+    meta_path = make_module(fname=fname, content=content)
+    monkeypatch.chdir(os.path.dirname(meta_path))
+    out = runner.invoke(cli, ["-n", "--no-build-isolation", "--find-links", fake_dists])
+    monkeypatch.undo()
+
+    assert out.exit_code == 0, out.stderr
+    assert "small-fake-a==0.1" in out.stderr
+    assert "small-fake-b" not in out.stderr
+    assert "small-fake-c" not in out.stderr
+    assert "extra ==" not in out.stderr
```

Comment:
```suggestion
    out = runner.invoke(
        cli,
        [
            "--output-file",
            "-",
            "--no-header",
            "--no-emit-options",
            "--no-annotate",
            "--no-build-isolation",
            "--find-links",
            fake_dists,
        ],
    )
    monkeypatch.undo()

    assert out.exit_code == 0, out.stderr
    assert "small-fake-a==0.1\n" == out.stdout
```

Could you please update the test so that the whole output file is checked?

Reply:
Sure, but should we do the same on `test_input_formats` too?

Reply:
Actually, there are multiple tests to update to use the same pattern. Feel free to fix it in a following up PR if you like.

Reply:
However, we can fix the current test here in this PR.

Reply:
I accidentally applied the suggestion twice and github broke the code :(